charts/zitadel/templates/deployment.yaml

```diff
@@ -95,6 +95,9 @@ spec:
                 - name: Host
                   value: {{ .Values.zitadel.configmapConfig.ExternalDomain }}
               {{- end }}
+              {{- if .Values.zitadel.configmapConfig.TLS.Enabled }}
```

Comment:
Doesn't this throw a nil pointer if TLS is undefined? I think there is a deepcheck helper function for this.

Reply:
> Doesn't this throw a nil pointer if TLS is undefined? I think there is a deepcheck helper function for this.

You are right. We should check for TLS first and then check for TLS.Enabled.

Reply:
This is resolved with https://github.com/zitadel/zitadel-charts/pull/144/commits/6dbd3f903b2f88b8ba11771bbac2908481d64097

Reply:
Unfortunately, this doesn't work for older Helm versions, because Helm <= 3.9 evaluates all arguments to *and* :-) This is why I added the deepCheck helper once and added this test run https://github.com/zitadel/zitadel-charts/blob/main/.github/workflows/test.yml#L40

Reply:
Got it.

Its changed to now use the deepCheck helper

`{{- if include "deepCheck" (dict "root" .Values "path" (splitList "." "zitadel.configmapConfig.TLS.Enabled")) }}`